### PR TITLE
add optional authentication

### PR DESCRIPTION
**Pull request by Bluekind1**
# Description

Added a toggle for authentication

# Issues

# Checklist

* [x] This PR contains a description of the changes I'm making
* [x] I updated the version in Chart.yaml
* [x] I updated the changelog with an `artifacthub.io/changes` annotation in `Chart.yaml`, check the [example](docs/development.md#Changelog) in the documentation.
* [x] I updated applicable README.md files using  `pre-commit run`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/vault-monitoring/templates/configmap.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if eq .Values.vault.serviceMonitor.authentication true }}
```
Comment (eyenx) [inline charts/vault-monitoring/templates/configmap.yaml:1]:
```suggestion
{{- if .Values.vault.serviceMonitor.authentication }}
```

File: charts/vault-monitoring/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -24,9 +24,11 @@ spec:
       port: {{ .Values.vault.portName }}
       scheme: {{ .Values.vault.scheme }}
       scrapeTimeout: {{ .Values.vault.serviceMonitor.scrapeTimeout }}
+    {{- if eq .Values.vault.serviceMonitor.authentication true }}
```
Comment (eyenx) [inline charts/vault-monitoring/templates/servicemonitor.yaml:27]:
```suggestion
    {{- if .Values.vault.serviceMonitor.authentication }}
```

File: charts/vault-monitoring/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -24,9 +24,11 @@ spec:
...
       {{- with .Values.vault.serviceMonitor.bearerTokenFile }}
       bearerTokenFile: {{ . | quote }}
```
Comment (eyenx) [inline charts/vault-monitoring/templates/servicemonitor.yaml:29]: We should revert @pree commit here. No need to check bearetoken is set if authentication is true

Comment (Bluekind1) [reply charts/vault-monitoring/templates/servicemonitor.yaml]: Reverted the change in the next commit

**Review by eyenx — APPROVED** [review]
LGTM. good work.

**Pull request merged**